src/utils/commandHandler.ts

```diff
@@ -0,0 +1,173 @@
+import {
+	Message,
+	Client,
+	BitFieldResolvable,
+	PermissionString,
+	MessageEmbed,
+	EmbedField,
+} from 'discord.js';
+
+interface commandHandlerOptions {
+	prefix: string;
+	logger: (...message: string[]) => void;
+	helpCommandFn?: (
+		commands: { aliases: string[]; description: string }[],
+		message: Message,
+	) => Promise<void>;
+	commandNotFoundFnOrMessage?:
+		| ((message: Message) => Promise<void> | void)
+		| string;
+}
+
+// Command Handler Class
+export class commandHandler {
+	private commands: Command[] = [];
+	private commandNotFoundFn: (message: Message) => Promise<void>;
+
+	constructor(private bot: Client, private option: commandHandlerOptions) {
+		// Logger
+		this.option.logger = option.logger || console.log;
+
+		// Not found command function
+		if (
+			!option.commandNotFoundFnOrMessage ||
+			typeof option.commandNotFoundFnOrMessage == 'string'
+		) {
+			this.commandNotFoundFn = async (message: Message) => {
+				await message.channel.send(
+					option.commandNotFoundFnOrMessage || 'Command Not Found',
+				);
+			};
+		}
+
+		// Add Help Command
+		this.addHelpCommand();
+
+		// Handle Message Event
+		this.bot.on('message', (message: Message) =>
+			this.handleMessageEvt(message),
+		);
+	}
+
+	/**
+	 * Register a command
+	 * @param command An Instance of Command Class
+	 */
+	registerCommand(command: Command): void {
```

Comment:
Should be private

Reply:
Uh. How will it register commands [here](https://github.com/SagnikPradhan/community-bot/blob/master/src/core/Client.ts#L31) ?

Reply:
ohh nvm